## Instructions {-}

1. Please answer the following questions as part of your project proposal.

2. Write your answers in the *Markdown* cells of the Jupyter notebook. You don't need to write any code, but if you want to, you may use the *Code* cells.

3. Use [Quarto](https://quarto.org/docs/output-formats/html-basics.html) to print the *.ipynb* file as HTML. You will need to open the command prompt, navigate to the directory containing the file, and use the command: `quarto render filename.ipynb --to html`. Submit the HTML file.

4. The project proposal is worth 8 points, and is due on **18th April 2023 at 11:59 pm**. 

5. You must make one submission as a group, and not individually.

6. Maintaining a GitHub repository is optional, though encouraged for the project.

7. Share the link of your project's GitHub repository [here](https://docs.google.com/spreadsheets/d/1khao3unpj_vsx4kOSg_Zzo77YK1UWL2w73Oa0aAirOo/edit#gid=0) (optional).

# 1) Team name
Mention your team name.

*(0 points)*

Zzz

# 2) Member names
Mention the names of your team members.

*(0 points)*

Diego Schummer, Aarti Pappu, Yasmeen Nahas, Divya Bhardwaj

# 3) Link to the GitHub repository (optional)
Share the link of the team's project repository on GitHub.

Also, put the link of your project's GitHub repository [here](https://docs.google.com/spreadsheets/d/1khao3unpj_vsx4kOSg_Zzo77YK1UWL2w73Oa0aAirOo/edit#gid=0).

We believe there is no harm in having other teams view your GitHub repository. However, if you don't want anyone to see your team's work, you may make the repository *Private* and add your instructor and graduate TA as *Colloborators* in it.

*(0 points)*

https://github.com/diegoschum/Zzz-303-3

In [2]:
import pandas as pd
#white wine 
white = pd.read_csv('winequality-white.csv', sep=';')
#red wine
red = pd.read_csv('winequality-red (1).csv', sep=';')


In [6]:
len(white)

4898

In [5]:
white.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.0              0.27         0.36            20.7      0.045   
1            6.3              0.30         0.34             1.6      0.049   
2            8.1              0.28         0.40             6.9      0.050   
3            7.2              0.23         0.32             8.5      0.058   
4            7.2              0.23         0.32             8.5      0.058   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 45.0                 170.0   1.0010  3.00       0.45   
1                 14.0                 132.0   0.9940  3.30       0.49   
2                 30.0                  97.0   0.9951  3.26       0.44   
3                 47.0                 186.0   0.9956  3.19       0.40   
4                 47.0                 186.0   0.9956  3.19       0.40   

   alcohol  quality  
0      8.8        6  
1      9.5        6  
2     10.1        6  
3      9.9        6  
4      9.9        6

# 4) Topic
Mention the topic of your course project.

*(0 points)*

# 5) Problem statement

*(4 points)*

Explain the problem statement. The problem statement must include:

## 5a) The problem

The goal of this project is to develop a machine learning model that can predict the quality of white wines based on their physicochemical properties.

## 5b) Type of response
Is it about predicting a continuous response or a binary response or a combination of both?

Predicting a continuous response – the quality of white wine is a continuous variable, measured on a scale from 0 to 10.



## 5c) Performance metric
How will you assess model accuracy?

  - If it is a classification problem, then which measure(s) will you optimize for your model – precision, recall, false negative rate (FNR), accuracy, ROC-AUC etc., and why?
  - If it is a regression problem, then which measure(s) will you optimize for your model – RMSE (Root mean squared error), MAE (mean absolute error), maximum absolute error etc., and why?

We will optimize the model for RMSE (Root Mean Squared Error) because it is a widely used and interpretable metric for regression problems. Minimizing RMSE will help us to develop a model that can accurately predict the quality of white wines.

## 5d) Naive model accuracy
What is the accuracy of the naive model (Standard deviation of response in case of continuous response / proportion of the majority class in case of classification model)

# 6) Data

## 6a) Source
What data sources will you use, and how will the data help solve the problem? Explain.
If the data is open source, share the link of the data.

*(0.5 point)*

The dataset is obtained from the UCI Machine Learning Repository, and it contains information about various physicochemical properties of white wines and their corresponding quality ratings. The dataset can be found at this link: https://archive.ics.uci.edu/ml/datasets/Wine+Quality

## 6b) Response & predictors
What is the response, and mention some of the predictors.

*(0.5 point)*

The response variable is the quality of the white wine, measured on a scale from 0 to 10. Some of the predictors include fixed acidity, volatile acidity, citric acid, residual sugar, chlorides, free sulfur dioxide, total sulfur dioxide, density, pH, and alcohol.

## 6c) Size
What is the number of continuous predictors, categorical predictors, and observations in your dataset(s). If you are using multiple datasets, please provide the information for each dataset. When counting predictors, count only those that have sufficient non-missing values, and will be useful.

*(1 point)*

The dataset contains 4,898 observations, with 11 continuous predictors and no categorical predictors.

# 7) Exisiting solutions
Are there existing solutions of your problem? Almost all Kaggle datasets have exisiting solutions. If yes, then how do you plan to build up on those solutions? **What is the highest model accuracy / performance achieved in the existing solutions?**

*(1 point)*

# 8) Stakeholders

Who are the stakeholders, and how will your project benefit them? Explain.

*(1 point)*

# 9) Work-split
*(This question is answered for you)*

How do you plan to split the project work amongst individual team members?

We will learn to develop and tune the following models in the STAT303 sequence:

1. MARS

2. Decision trees with cost-complexity pruning

3. Bagging (Bagging MARS / decision trees)

4. Random Forests

5. AdaBoost

6. Gradient boosting

7. XGBoost

8. Lasso / Ridge / Stepwise selection 

Each team member is required to develop and tune at least one of the above models. In the end, the team will combine all the developed models to create a model more accurate than each of the individual models.

*(0 points)*